firefox-ios/Client/Frontend/Autofill/Address/AddressCellView.swift

```diff
@@ -0,0 +1,88 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import SwiftUI
+import Common
+import Shared
+import Storage
+
+// MARK: - AddressCellView
+
+/// A view representing a cell displaying address information.
+struct AddressCellView: View {
+    // MARK: - Properties
+
+    @State private var textColor: Color = .clear
+    @State private var customLightGray: Color = .clear
+    @State private var iconPrimary: Color = .clear
+
+    private(set) var address: Address
+    @Environment(\.themeType)
+    var themeVal
+    private(set) var onTap: () -> Void
+
+    // MARK: - Body
+
+    var body: some View {
+        Button(action: onTap) {
+            VStack(alignment: .leading, spacing: 0) {
+                HStack(spacing: 24) {
+                    Image(ImageIdentifiers.locationAddressAutofillIcon)
+                        .renderingMode(.template)
+                        .padding(.leading, 16)
+                        .foregroundColor(iconPrimary)
+                        .offset(y: -14)
+                    VStack(alignment: .leading) {
+                        Text(address.givenName + " " + address.familyName)
+                            .font(.body)
+                            .foregroundColor(textColor)
+                        Text(address.streetAddress)
+                            .font(.subheadline)
+                            .foregroundColor(customLightGray)
+                        Text(address.addressLevel2 + ", " + address.addressLevel1 + " " + address.postalCode)
```

Comment:
This is something that should be in your view model and testable @jnrahme 🤔

Reply:
Updated